cmds/gzip/gzip.go

```diff
@@ -0,0 +1,454 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/klauspost/pgzip"
+	"github.com/spf13/pflag"
+)
+
+const version string = "0.0.1"
+
+type options struct {
+	blocksize  int
+	level      int
+	processes  int
+	decompress bool
+	force      bool
+	help       bool
+	keep       bool
+	license    bool
+	//list       bool
+	//name       bool
+	//noName     bool
+	quiet bool
+	//recursive  bool
+	//rsyncable  bool
+	stdout bool
+	//test       bool
+	verbose bool
+	version bool
+	suffix  string
+}
+
+func (o *options) parseArgs() error {
+	var levels [10]bool
+
+	pflag.IntVarP(&o.blocksize,
+		"blocksize",
+		"b",
+		128,
+		"Set compression block size in KiB")
+
+	pflag.BoolVarP(&o.decompress,
+		"decompress",
+		"d",
+		false,
+		"Decompress the compressed input")
+
+	pflag.BoolVarP(&o.force,
+		"force",
+		"f",
+		false,
+		"Force overwrite of output file and compress links")
+
+	pflag.BoolVarP(&o.help,
+		"help",
+		"h",
+		false,
+		"Display a help screen and quit")
+
+	pflag.BoolVarP(&o.keep,
+		"keep",
+		"k",
+		false,
+		"Do not delete original file after processing")
+
+	pflag.BoolVarP(&o.license,
+		"license",
+		"L",
+		false,
+		"Display license")
+
+	/*
```

Comment:
I'm not a big fan of commented out code, why not just leave a TODO here with 
// TODO: implement list and name here
// no-name makes no sense to me , it's just the opposite of name right? Why need two?

Reply:
// TODOs are a reasonable approach.

It appears --name is default on compression,

> When compressing, always save the original file name and time stamp; this is the default. When decompressing, restore the original file name and time stamp if present. This option is useful on systems which have a limit on file name length or when the time stamp has been lost after a file transfer.


and --no-name is default behavior on decompression,

> When compressing, do not save the original file name and timestamp by default. (The original name is always saved if the name had to be truncated.) When decompressing, do not restore the original file name if present (remove only the gzip suffix from the compressed file name) and do not restore the original time stamp if present (copy it from the compressed file). This option is the default when decompressing.

Both options are needed to counteract the defaults it appears. That being said, looking through klauspost/pgzip, I don't see a straight forward way to implement either of these options to negate the default behaviors noted above. I also don't actually see the use case of negating either of the default behaviors.